### ISSUE TITLE
UB in ping task: pointer passed to read_volatile must not be null

### ISSUE THREAD

**Issue by saethlin** [oxidecomputer/hubris#276]
This code executes undefined behavior:
https://github.com/oxidecomputer/hubris/blob/8833cc1dcfdbf107369e188a035a38651ba7ab46/task/ping/src/main.rs#L19

The `std::ptr` module documents:
> A null pointer is never valid, not even for accesses of size zero.

And `ptr::read_volatile` repeats this:
> Note that even if T has size 0, the pointer must be non-null and properly aligned.

I think this line should be replaced with inline assembly that does the faulting read.

**Comment by cbiffle** [comment]
You are correct that this is UB. This task is also kind of a placeholder hack, which was originally written to demonstrate that faults worked. We now have a test suite for that, so within the team we're of two minds here:

1. Just remove `task/ping`. (This is my preferred approach.)
2. Retain `task/ping` as a demo, since it makes for a nice humility tutorial, tracking down the various issues. I see the merit of this; in this case we'd probably want to rewrite the code to, uh, correctly dereference a null pointer.

Note that there's another option for making this code correct, which is to change it to `1`. ;-)
